core/src/main/java/com/linecorp/armeria/client/endpoint/EndpointSelectionTimeoutException.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * An {@link EndpointGroupException} raised when an {@link EndpointGroup} fails to resolve
+ * an {@link Endpoint} within a configured selection timeout.
+ */
+public final class EndpointSelectionTimeoutException extends EndpointGroupException {
```

Comment:
Might be a better idea to extend from `EmptyEndpointGroupException` since:
1) It can be argued this is a type of `EndpointGroupException`
2) Users experience fewer breaking changes since they are already catching this exception.